src/components/AddDevice/AddDevice.tsx

```diff
@@ -1,9 +1,188 @@
-import * as React from 'react';
+import { createKey } from '@near-js/biometric-ed25519';
+import { sendSignInLinkToEmail } from 'firebase/auth';
+import React, { useEffect } from 'react';
+import { useForm } from 'react-hook-form';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import styled from 'styled-components';
+
+import { Button } from '../../lib/Button';
+import { openToast } from '../../lib/Toast';
+import { firebaseAuth } from '../../utils/firebase';
+import { isValidEmail } from '../../utils/form-validation';
+
+const handleCreateAccount = async ({
+  accountId, email, isRecovery, success_url, failure_url, public_key, contract_id, methodNames
+}) => {
+  const keyPair = await createKey(email);
+  const publicKeyWebAuthn = keyPair.getPublicKey().toString();
+
+  if (!publicKeyWebAuthn) {
+    throw new Error('No public key found');
+  }
+
+  const searchParams = new URLSearchParams({
+    publicKey: publicKeyWebAuthn,
+    email,
+    ...(accountId ? { accountId } : {}),
+    ...(isRecovery ? { isRecovery: 'true' } : {}),
+    ...(success_url ? { success_url } : {}),
```

Comment:
@MaximusHaximus can this interface change be pushed forward? I'm currently using [almost a duplicate of `WalletConnection` from NAJ](https://github.com/near/near-discovery/compare/fast-auth-iframe-wallet#diff-59de94921f8f61e171c1f1c3c01c0ecdad0eb610266c2f0d67d61a05813e02c7R31) to interface with the signer app form dapps and would prefer to keep the interface consistent with that to avoid having to deviate from that code too much and test it separately